MantidQt/CustomInterfaces/src/Reflectometry/ReflDataProcessorPresenter.cpp

```diff
@@ -305,9 +305,8 @@ void ReflDataProcessorPresenter::parseUniform(const std::string &timeSlicing,
     return;
   }
 
-  const auto minTime = mws->getFirstPulseTime();
-  const auto maxTime = mws->getLastPulseTime();
-  const auto totalDuration = maxTime - minTime;
+  const auto run = mws->run();
+  const auto totalDuration = run.endTime() - run.startTime();
   double totalDurationSec = totalDuration.seconds();
```

Comment:
This is wrong, it should be

`double totalDurationSec = totalDuration.total_seconds();`

as suggested in a comment above. Having just `seconds()` does not produce the correct results.

In general, when working on a new piece of work/fixing something you should think about the best way of checking that results are correct. In this case it is very easy, you should check that the following three cases give the same `IvsQ` and `IvsLam` workspaces:

1. Custom slicing with Python list `0, 122, 244, 366, 488, 610`
2. Uniform slicing with `5` even time slices
3. Uniform slicing with `122` seconds

The reason is that run `INTER38415` has a total duration of `610` seconds (which is something you can see from the logs), and `610` is divisible by `5`, which means that the above cases should produce the same time slices. Check this when you change the line above and you'll see that, for instance, the `IvsQ` workspaces match.